terminal.go

```diff
@@ -764,23 +797,39 @@ func (g *LightningTerminal) setUpLNDClients() error {
 	}()
 
 	log.Infof("Connecting full lnd client")
-	g.lndClient, err = lndclient.NewLndServices(
-		&lndclient.LndServicesConfig{
-			LndAddress:            host,
-			Network:               network,
-			TLSPath:               tlsPath,
-			Insecure:              insecure,
-			CustomMacaroonPath:    macPath,
-			CustomMacaroonHex:     hex.EncodeToString(macData),
-			BlockUntilChainSynced: true,
-			BlockUntilUnlocked:    true,
-			CallerCtx:             ctxc,
-			CheckVersion:          minimalCompatibleVersion,
-		},
-	)
-	if err != nil {
-		return fmt.Errorf("could not create LND Services client: %v",
-			err)
+	for {
+		g.lndClient, err = lndclient.NewLndServices(
+			&lndclient.LndServicesConfig{
+				LndAddress:            host,
+				Network:               network,
+				TLSPath:               tlsPath,
+				Insecure:              insecure,
+				CustomMacaroonPath:    macPath,
+				CustomMacaroonHex:     hex.EncodeToString(macData),
+				BlockUntilChainSynced: true,
```

Comment:
Since we are passing in `BlockUntilChainSynced: true` here, we don't actually need to re-try ourselves, as that will already initiate a re-try loop, as long as the `ctxc` context we pass in doesn't expire too early.

Reply:
Same goes for the `BlockUntilUnlocked: true` I guess. So maybe we could also just try to initiate the full client first, as that already does all the re-trying we want and need and then can initiate the basic client afterwards where we can be pretty certain that it should succeed?

Reply:
I guess what we lose that way would be the extra info exposed to the user with the `SetErrored` field. So I'm fine with leaving the order the way it is since it gives the user more information. But I'm fairly certain we don't need the re-try `for` loop here as well.

Reply:
> Since we are passing in BlockUntilChainSynced: true here, we don't actually need to re-try ourselves, as that will already initiate a re-try loop, as long as the ctxc context we pass in doesn't expire too early.

So unfortunately this isn't really the case, and logs we have seen from users, such as @Liongrass, have had the `ctxc` time out during the chain sync of `lnd`, when `lnd` took for example 2mins+ to successfully sync. The `GetInfo` call that's made during the chain sync in the `lndclient` then times out, and therefore the set up of the full client fails. If we would have retried to set up the full `lndclient` during those cases, the `lnd` sub-server in litd would have eventually successfully been started.
One potential alternative solution could have been to increase the `ctxc` timeout, but as there's no specific context just for the chain sync, that would mean that any future RPC call made after the chain sync would also use the updated timeout, which wouldn't be ideal..

This is why we decided to take the approach to attempt to set up the `lndclient` indefinitely until it is successful as a temporary fix, until we implement a more general solution that restarts `litd` sub-servers when required to restart.

Reply:
Ah right, good point. Makes sense.